File: hw/aimacana/u3/POO/java/rpg-system/src/main/java/ec/edu/espe/rpg/model/entities/Warrior.java
package ec.edu.espe.rpg.model.entities;

public class Warrior extends Character {
    private double strength;
    private double rage;

    public Warrior(String id, String name, int level, double maxHp, double strength) {
        super(id, name, level, maxHp);
        this.strength = strength;
        this.rage = 0;
    }

    public double getStrength() { return strength; }
    public void setStrength(double strength) { this.strength = strength; }

    public double getRage() { return rage; }

    @Override
    public void attack(Character target) throws ec.edu.espe.rpg.model.exceptions.CharacterDeadException {
        if (this.getHp() <= 0) {
            throw new ec.edu.espe.rpg.model.exceptions.CharacterDeadException("El guerrero está muerto y no puede atacar.");
        }
        double damage = (strength * 1.5) + getBonusDamage();
        target.takeDamage(damage);
        this.rage += 10;
        if (this.rage > 100) this.rage = 100;
    }
}

File: hw/aimacana/u3/POO/typescript/rpg-system/src/server/models/Warrior.ts
import { Character } from './Character';
import { CharacterDeadException } from './exceptions';

export class Warrior extends Character {
  private strength: number;
  private rage: number;

  constructor(id: string, name: string, level: number, maxHp: number, strength: number) {
    super(id, name, level, maxHp);
    this.strength = strength;
    this.rage = 0;
  }

  public getStrength(): number { return this.strength; }
  public setStrength(strength: number): void { this.strength = strength; }

  public getRage(): number { return this.rage; }

  public attack(target: Character): void {
    if (this.getHp() <= 0) {
      throw new CharacterDeadException('El guerrero está muerto y no puede atacar.');
    }
    const damage = (this.strength * 1.5) + this.getBonusDamage();
    target.takeDamage(damage);
    this.rage += 10;
    if (this.rage > 100) this.rage = 100;
  }
}
